hotsos/core/ycheck/engine/properties/requires/types/apt.py

```diff
@@ -28,32 +28,147 @@ def installed_versions(self):
 
         return _versions
 
-    def package_version_within_ranges(self, pkg, versions):
-        result = False
-        versions = sorted(versions, key=lambda i: str(i['min']), reverse=True)
+    def normalize_version_criteria(self, version_criteria):
+        """Normalize all the criterions in a criteria.
+
+        Normalization does the following:
+        - removes empty criteria
+        - replaces old ops with the new ones
+        - sorts each criterion(ascending) and criteria(descending)
+        - adds upper/lower bounds to criteria, where needed
+
+        Args:
```

Comment:
I do have a plan to switch the codebase to Google style doctrings (I agree they are better) but for now lets please stick to the existing convention (javadoc) i.e. ```@param <name>: <description>, @return: <description>```